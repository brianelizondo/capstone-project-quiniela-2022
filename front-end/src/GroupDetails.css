.GroupDetails-subtitle{
    color: white;
    font-weight: 100 !important;
    font-size: 18px;
    background-color: rgba(0,0,0, .60);
    display:inline-block;
    padding: 15px 15px;
    border-left: 5px solid black;
    margin-top: 50px;
    margin-bottom: 20px;
}

.GroupDetails-matches a{
    text-decoration: none;
    color: black;
}