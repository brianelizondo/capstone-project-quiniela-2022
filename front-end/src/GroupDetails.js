import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Row, Col } from 'react-bootstrap';
import Loading from './Loading';
import './GroupDetails.css';

// import APIFootball api
import APIFootball from "./api-football";

import GroupStandings from './GroupStandings';
import MatchCard from './MatchCard';

function GroupDetails(){
    // initial state
    const { group } = useParams();
    const [loading, setLoading] = useState(false);
    const [groupsStandings, setGroupsStandings] = useState([]);
    const [groupMatches, setGroupMatches] = useState([]);
    const [matchesGoals, setMatchesGoals] = useState({});

    useEffect(() => {
        setLoading(true);
        async function getGroupsStandings() {
            const resp = await APIFootball.getGroupsStandings();
            setGroupsStandings(resp);
        }
        getGroupsStandings();

        async function getGroupMatches() {
            const resp = await APIFootball.getGroupMatches(group);
            setGroupMatches(resp);

            const goalsAPI = await APIFootball.getMatchesGoals();
            setMatchesGoals(goalsAPI);
        }
        getGroupMatches();
        setLoading(false);
    }, [group]);

    if(loading){
        return <Loading />;
    }

    return (
        <div className="GroupDetails col-md-8 offset-md-2">
            <Row>
                <Col>
                    <h1 className='section-title'>Details of the Group "{group}"</h1>
                </Col>
            </Row>
            <Row>
                <Col>
                    <h5 className='GroupDetails-subtitle'>Group Standings</h5>
                    <div className="GroupDetails-standings">
                        <GroupStandings key={group} group={group} standings={ groupsStandings.filter(team => team.group === group) } detailsButton={false} />
                    </div>
                </Col>
            </Row>
            <Row>
                <Col>
                    <h5 className='GroupDetails-subtitle'>Group Matches</h5>
                    <div className="GroupDetails-matches">
                        { groupMatches.map(match => <MatchCard key={match.id} match={match} matchesGoals={matchesGoals} />) }
                    </div>
                </Col>
            </Row>
        </div>
    );
}

export default GroupDetails;