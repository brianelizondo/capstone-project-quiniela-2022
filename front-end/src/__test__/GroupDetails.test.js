import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import GroupDetails from '../GroupDetails';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'GroupDetails' component without crashing", () => {
        render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupDetails />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Group Standings");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'GroupDetails' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupDetails />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Additional Test", () => {
    it("works when you click on a team name go to the team details", async () => {
        render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupDetails />
            </MemoryRouter>
        );
        const user = userEvent.setup()

        waitFor(() => { expect(screen.getByText('Details for group "A"')).toBeInTheDocument() });
        waitFor(async () => { await user.click(screen.getAllByTestId("team-")) });
        waitFor(() => { expect(screen.getByText("Details for the team")).toBeInTheDocument() });
    });

    it("works when you click on a match card go to the match details", async () => {
        render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupDetails />
            </MemoryRouter>
        );
        const user = userEvent.setup()

        waitFor(() => { expect(screen.getByText('Details for group "A"')).toBeInTheDocument() });
        waitFor(async () => { await user.click(screen.getAllByTestId("match-")) });
        waitFor(() => { expect(screen.getByText("Details and Statistics about the match")).toBeInTheDocument() });
    });
});