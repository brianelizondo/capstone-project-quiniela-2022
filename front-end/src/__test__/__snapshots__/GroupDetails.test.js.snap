// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'GroupDetails' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GroupDetails col-md-8 offset-md-2"
  >
    <h1>
      Groups in FIFA World Cup 2022
    </h1>
    <p>
      Details for group ""
    </p>
    <h4>
      Group Standings
    </h4>
    <div
      class="GroupDetails-standings"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group 
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
    <h4>
      Group Matches
    </h4>
    <div
      class="GroupDetails-matches"
    />
  </div>
</DocumentFragment>
`;
